x/delayedack/keeper/finalize.go

```diff
@@ -3,36 +3,79 @@ package keeper
 import (
 	"fmt"
 
-	"github.com/cosmos/ibc-go/v7/modules/core/exported"
-
-	"github.com/cometbft/cometbft/libs/log"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
 	porttypes "github.com/cosmos/ibc-go/v7/modules/core/05-port/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
 	"github.com/osmosis-labs/osmosis/v15/osmoutils"
 
 	commontypes "github.com/dymensionxyz/dymension/v3/x/common/types"
 	"github.com/dymensionxyz/dymension/v3/x/delayedack/types"
 )
 
 // FinalizeRollappPackets finalizes the packets for the given rollapp until the given height which is
-// the end height of the latest finalized state
-func (k Keeper) FinalizeRollappPackets(ctx sdk.Context, ibc porttypes.IBCModule, rollappID string, stateEndHeight uint64) error {
+// the end height of the latest finalized state. Returns the number of finalized packets.
+func (k Keeper) FinalizeRollappPackets(ctx sdk.Context, ibc porttypes.IBCModule, rollappID string, stateEndHeight uint64) (int, error) {
+	// Get the latest state info of the rollapp
+	latestIndex, found := k.rollappKeeper.GetLatestFinalizedStateIndex(ctx, rollappID)
+	if !found {
+		return 0, fmt.Errorf("latest finalized state index for rollapp '%s' is not found", rollappID)
```

Comment:
should use gerrc package (gerrc.ErrNotFound)

Also, don't embed data in error strings